src/hooks/use-menu/index.tsx

```diff
@@ -20,6 +20,19 @@ export const useMenu = (): useMenu => {
       'aria-haspopup': true,
       'aria-expanded': isOpen,
       role: 'button',
+      onKeyDown: (e) => {
+        if (e.key === 'ArrowDown') {
+          const menuItems = e.currentTarget.parentElement!.querySelectorAll(
+            '[role="menuitem"]',
+          ) as NodeListOf<HTMLElement>
+
+          if (menuItems.length) menuItems[0].focus()
+        }
+
+        if (props?.onKeyDown) {
+          props.onKeyDown(e)
+        }
```

Comment:
**note:** Not a big deal, but we can write `props?.onKeyDown?.(e)` instead of needing the if-statement to assert the function is defined before calling. That is, the optional chaining operator (`?.`) can apply to function calls, not just property access.

Reply:
thanks for the tips 👍